codebundles/k8s-argocd-helm-health/runbook.robot

```diff
@@ -0,0 +1,98 @@
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    \n\nArgoCD Helm releases available: \n\n ${helmreleases.stdout}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Check ArgoCD Helm releases Status
+    [Documentation]    Check ArgoCD helm release Status.
+    [Tags]    argocd    helmrelease    status    health
+    ${argocd_helm_status}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} -n ${NAMESPACE} --context ${CONTEXT} -o=json | jq -r '.items[] | "Name:\\t\\t\\t" + .metadata.name + "\\nHealth Status:\\t\\t" + .status.health.status + "\\nSync Status:\\t\\t" + .status.sync.status + "\\nOperational State:\\t" + .status.operationState.message'
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    \n\nArgoCD Helm release Status: \n\n ${argocd_helm_status.stdout}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+
+*** Keywords ***
+Suite Initialization
+    ${kubeconfig}=    RW.Core.Import Secret
+    ...    kubeconfig
+    ...    type=string
+    ...    description=The kubernetes kubeconfig yaml containing connection configuration used to connect to cluster(s).
+    ...    pattern=\w*
+    ...    example=For examples, start here https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/
+
+    ${kubectl}=    RW.Core.Import Service    kubectl
+    ...    description=The location service used to interpret shell commands.
+    ...    default=kubectl-service.shared
+    ...    example=kubectl-service.shared
+
+    ${DISTRIBUTION}=    RW.Core.Import User Variable    DISTRIBUTION
+    ...    type=string
+    ...    description=Which distribution of Kubernetes to use for operations, such as: Kubernetes, OpenShift, etc.
+    ...    pattern=\w*
+    ...    enum=[Kubernetes,GKE,OpenShift]
+    ...    example=Kubernetes
+    ...    default=Kubernetes
+
+    ${KUBERNETES_DISTRIBUTION_BINARY}=    RW.Core.Import User Variable    KUBERNETES_DISTRIBUTION_BINARY
+    ...    type=string
+    ...    description=Which binary to use for Kubernetes CLI commands.
+    ...    enum=[kubectl,oc]
+    ...    example=kubectl
+    ...    default=kubectl
+
+    ${NAMESPACE}=    RW.Core.Import User Variable
+    ...    NAMESPACE
+    ...    type=string
+    ...    description=The name of the Kubernetes namespace to scope actions and searching to. Accepts a single namespace in the format `-n namespace-name` or `--all-namespaces`.
+    ...    pattern=\w*
+    ...    example=-n my-namespace
+    ...    default=--all-namespaces
+
+    ${RESOURCE_NAME}=    RW.Core.Import User Variable
```

Comment:
We might just choose to get rid of this variable, or set the default to applications.argoproj.io. - your call.

Reply:
@stewartshea Agreed and updated in latest commit